### PR TITLE
Add watsonx.ai feature

### PR DESCRIPTION
**Pull request by PabloSanchi**
# Add IBM watsonx.ai support

### CODE REVIEW

**Comment by PabloSanchi** [comment]
This repo shows how to use it, In case it comes in handy: https://github.com/PabloSanchi/IBM-WatsonxAI-Spring-AI-Example

**Comment by tzolov** [comment]
Hey @PabloSanchi , great contribution! Thanks. 
I will review it shortly

**Comment by markpollack** [comment]
Thanks so much for this.  Will review and learn more about watstonx.ai!

File: spring-ai-spring-boot-autoconfigure/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports [hunk 1]
```diff
@@ -23,4 +23,5 @@ org.springframework.ai.autoconfigure.vectorstore.weaviate.WeaviateVectorStoreAut
 org.springframework.ai.autoconfigure.vectorstore.neo4j.Neo4jVectorStoreAutoConfiguration
```
Comment (yaoyaoyao480) [inline spring-ai-spring-boot-autoconfigure/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports:23]: 评论

**Comment by tzolov** [comment]
Additional pre-merge adjustments applied:
    
     - Rename all WatsonxAIXxx classes to WatsonxAiXxx.
     - Rename WatsonxChatClient to WatsonxAiChatClient.
     - Move WatsonxAiChatOptions out of the API.
     - Implement a Builder for WatsonxAiChatOptions (replace the inline withXxx code).
     - Add a WatsonxAiChatOptions field to WatsonxAiChatClient as default options.
       Later, it is also set by the auto-configuration properties.
     - Implement merging logic for default vs runtime options in WatsonxAiChatClient.
     - In Auto-config, add WatsonxAiChatProperties with enabled and options fields.
       Options are passed to the client.
     - Update the adoc to include the .chat.options properties.
     - Add the watsonxai doc to the nav.adoc.
     - Fix license headers and javadocs.
     - Move dependency versioning to the parent POM.

Then rebased, squashed and merged at 52d15f5dc1acc2a6d884fb0348dc5788971fdea4

**Comment by PabloSanchi** [comment]
@tzolov thank you for your time and consideration I appreciate it. 
Perfect, I will check it as soon as possible and give feedback about the results :)

**Pull request closed without merging**